exporters/otlp/CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 include_directories(include)
 
 add_library(opentelemetry_exporter_otprotocol src/recordable.cc)
+target_include_directories(
+  opentelemetry_exporter_otprotocol
+  PUBLIC $<TARGET_PROPERTY:opentelemetry_proto,INTERFACE_INCLUDE_DIRECTORIES>)
 target_link_libraries(opentelemetry_exporter_otprotocol
                       $<TARGET_OBJECTS:opentelemetry_proto>)
 
```

Comment:
For the tests below, could you please add `otlp_exporter_test` in addition to `recordable_test`? That should now link properly with the gRPC dependency available.

Reply:
Sorry for the long delay. I've tried to add the tests, but I found that gRPC actually uses the same flag to enable tests as in this project. That gave me a problem since one of gRPC's tests didn't build correctly. I currently workaround it by renaming the OT flag to BUILD_TESTING_OT. Next it turned out that need to generate more gRPC code for testing, but unfortunately I haven't had time to complete that yet due to lot's of other work. Hopefully will get better time in the coming weeks to get this completed.

Any thoughts on the flag to enable tests since it clashes with gRPC? Since it slows down the build having to also test gRPC and its dependencies it certainly has disadvantages. But it can also be considered as a good thing to know that gRPC works correctly.

Reply:
I think gRPC uses [`grpc_BUILD_TEST`](https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L47). In any case, we should avoid running gRPC tests.

If it's necessary, I have no qualms with renaming `BUILD_TESTING` and adding a OT specific prefix or postfix.

Reply:
I think you're right, @pyohannes. It's a Abseil, a dependency to gRPC, that uses BUILD_TESTING too.
Unfortunately, I still can't tests to be built. It appears to be a clash between which protobuf compiler I've got installed on my system and what is included in gRPC added with CMake.

At the top of the generated header files for gRPC it adds:
```
#include <google/protobuf/port_def.inc>
#if PROTOBUF_VERSION < 3011000
#error This file was generated by a newer version of protoc which is
#error incompatible with your Protocol Buffer headers. Please update
#error your headers.
#endif
#if 3011002 < PROTOBUF_MIN_PROTOC_VERSION
#error This file was generated by an older version of protoc which is
#error incompatible with your Protocol Buffer headers. Please
#error regenerate this file with a newer version of protoc.
#endif
```

One workaround is to also add Protobuf with FetchContent, but I found a project that struggled to make that approach work https://github.com/google/or-tools/issues/2219 which links to a CMake issue at https://gitlab.kitware.com/cmake/cmake/-/issues/17735

Reply:
I wonder how this clash between protobuf versions comes about. In such a scenario, we should always use the one added with gRPC via CMake (by adding local paths, include paths and library paths before system paths).